transport/tcp/serve_tcp_ipmap.go

```diff
@@ -60,38 +46,62 @@ func (m *ipMap) Get(ipAddr *net.IPAddr) (string, error) {
 			return e.ident, nil
 		}
 		if e.subnet != nil {
-			contains := e.subnet.Contains(ipAddr.IP)
-			if contains {
-				if e.ident == "willbeclientip" {
-					return ipAddr.IP.String(), nil
-				}
-				return e.ident + ipAddr.IP.String(), nil
+			if e.subnet.Contains(ipAddr.IP) {
+				return strings.Replace(e.ident, "*", ipAddr.IP.String(), 1), nil
```

Comment:
maybe put an assertion or at least a comment here that it's guaranteed that e.ident will contain at most one "*"